library/Zend/Code/Generator/ValueGenerator.php

```diff
@@ -239,7 +318,7 @@ public function generate()
                 );
                 foreach ($rii as $curKey => $curValue) {
                     if (!$curValue instanceof ValueGenerator) {
-                        $curValue = new self($curValue);
+                        $curValue = new self($curValue, self::TYPE_AUTO, self::OUTPUT_MULTIPLE_LINE, $this->getConstants());
```

Comment:
@3axap4eHko Did you use ArrayObject for this line?

I don't understand the performance improvements you claim. Non return by reference methods implemented on StdLib\ArrayObject are invoked.

The only thing I see the constants are shared by the instances created in this line, so the only need I see $this->constants must be provided as reference. Alternative solutions may

* Use PHP native `ArrayObject`
* Refactor getConstants for return by reference.